### ISSUE TITLE
Alternative scale zero property (in addition to true or false)

### ISSUE THREAD

**Issue by kanitw** [vega/vega#1471]
From https://twitter.com/Datawrapper/status/1070702878061416448, 

![image](https://user-images.githubusercontent.com/111269/49667739-ccf79b00-fa10-11e8-80ff-25843e9a77db.png)

I could see we extend the scale's `zero` property with an object describing automatic criteria such as maximum % that could be added.

@jheer -- I wonder where is the logic for reading scale's `zero` and adding it to the domain in the codebase?  (I searched for "zero" in [vega-scale](https://github.com/vega/vega-scale/search?q=zero&unscoped_q=zero) and [vega-parser](https://github.com/vega/vega-parser/search?q=zero&unscoped_q=zero) but couldn't find one.)

**Comment by domoritz** [comment]
Couldn't you just write an expression that determines the domain based on how far away it is from zero?

**Comment by kanitw** [comment]
Would the monorepo in https://github.com/vega/vega/pull/1472 land any time soon?  

If I have time over the holiday break, I might try to fix this. :)

**Comment by jheer** [comment]
Yes, I plan to merge the monorepo to master sometime in the next week. Everything is working, just want to double check everything and update the documentation / README.md files.

**Comment by kanitw** [comment]
Ah I was playing with this and found this issue. (FWIW, this is a similar [Vega-Lite spec for playing with the threshold](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEzjQATjRyZ289AEEABNCTY46nILaZjMNgHcassobvqIcHA-OGks5YYINIJqhjg+FIJWshCGAGZs6obESJpsDLFQqRCUiiA4qUgIEGgA2qCyRXBoIEGyAPooSimCDNWoACwADF1KTE7KaKBOOAyYNep+ZNVKdWj8AIw9vbVIAB5oSz0AvtsKFVU1yGuNeS1tm8t9A0MgI2MTUzO1TvNbK8eXO3sHCO0gAC9zGw8nA1jh1DUkEwIAAKY6NQz8QJORoASgIx1h0LhCKQaIUvhh8NR+LRhgAfABeQxdSgANhA2wAukplKYUKhQB4AI5tWRQdrczCpcaoFYQTBsHBfLq7JSYSYxBLqBBlUBZQRQBiCUz-eHrRHIurowwAKkMSuUbAQsPJAGoUQ0miAkCV0CgfiBNdrdZh-labXa8m6akwmazVuoANY1LxOcZKOACtjKJxkW4bLkgOI0OCCQYevKYACe4RqfL8OhFOlIeRMIVuQPUIOzYIhNWbIN2exAJduufzhZA4YVZf+ldk1dMNDrPaAA))

cc: @yhoonkim
